components/concept/sync/src/main/java/mozilla/components/concept/sync/Devices.kt

```diff
@@ -0,0 +1,147 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+@file:SuppressWarnings("TooManyFunctions")
+package mozilla.components.concept.sync
+
+import androidx.lifecycle.LifecycleOwner
+import kotlinx.coroutines.Deferred
+import mozilla.components.support.base.observer.Observable
+
+/**
+ * Describes available interactions with the current device and other devices associated with an [OAuthAccount].
+ */
+interface DeviceConstellation : OAuthAccount, Observable<DeviceEventsObserver> {
+    /**
+     * Make sure there is an available a device record in the [DeviceConstellation] for the current device.
+     * @param name An initial name for the current device. This may be changed via [setDeviceNameAsync].
+     * @param type Type of the current device. This can't be changed.
+     * @param capabilities A list of capabilities that the current device claims to have.
+     */
+    fun initDeviceAsync(
+        name: String,
+        type: DeviceType = DeviceType.MOBILE,
+        capabilities: List<DeviceCapability>
+    ): Deferred<Unit>
+
+    /**
+     * Ensure that all initialized capabilities are configured.
```

Comment:
nit: would be nice to elaborate a bit here. What are capabilities? Probably just link to `[DeviceCapability]`